msdm/tests/test_laostar.py

```diff
@@ -41,5 +42,19 @@ def heuristic(s):
         print(traj.state_traj)
         assert traj.state_traj[-1] == goal
         
+    def test_trivial_solution(self):
+        algo = LAOStar(seed=42)
+        # Normal
+        mdp = Counter(3, initial_state=0)
+        R = algo.plan_on(mdp)
+        assert R.sGraph[mdp.initial_state()]['value'] == -3
+        assert R.policy.run_on(mdp).action_traj == (+1, +1, +1)
+
+        # No-op task. Now we start at 3, so value should be 0 there
+        mdp = Counter(3, initial_state=3)
+        R = algo.plan_on(mdp)
+        assert R.sGraph[mdp.initial_state()]['value'] == 0
```

Comment:
test fails with value of -103, and `run_on` seems to infinite loop, so I think it doesn't recognize that the initial state can be terminal too.

Reply:
this should be fixed now - it wasn't handling the semantics of `is_terminal` correctly when the initial state is the terminal state. formally, a terminal state is a state for which all actions lead back to the terminal state w.p. 1 and the reward is 0.